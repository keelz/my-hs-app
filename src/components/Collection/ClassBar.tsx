import * as React from 'react';
import * as classnames from 'classnames';
import { IComponentProps } from '../../common/models/App.model';
import { NavBar } from '../NavBar';
import { composeClassname } from '../../common/utils';
import NavItem from '../NavBar/NavItem';

interface IClassBarProps extends IComponentProps {}

export interface IClassBarStateProps {
  activeClassName?: string;
  cardClassNames: string[];
  setActiveCardClassName: (cardClassName: string) => any;
}

type Props = IClassBarProps & IClassBarStateProps;

export const composeButtonClassname = (
  className: string,
  activeClassName?: string
) => !!activeClassName && activeClassName === className
  ? ['btn btn-light active']
  : ['btn btn-light'];

const ClassBar: React.SFC<Props> = props =>
  <NavBar
    className={composeClassname('Collection-classbar')(props.className)}
    id="card-class-name-nav-bar">
    { props.cardClassNames.map(cardClassName =>
        <NavItem
          key={`card-class-nav-link-${cardClassName}`}
          render={() => {
            return (
              <button
                className={classnames(composeButtonClassname(
                  cardClassName,
                  props.activeClassName
                ))}
                onClick={() => props.setActiveCardClassName(cardClassName)}>
                {cardClassName}
              </button>
            );
          }} />
    )}
  </NavBar>;

export default ClassBar;
