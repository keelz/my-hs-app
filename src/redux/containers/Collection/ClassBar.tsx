import { connect } from 'react-redux';
import { IRootState } from '../../Types';
import { collectionSetActiveClassNameAction } from '../../reducers/Collection';
import { selectCardClassNames } from '../../selectors/Collection';
import ClassBar from '../../../components/Collection/ClassBar';

const mapStateToProps = (state: IRootState) => ({
  activeClassName: state.Collection.activeClassName,
  cardClassNames: selectCardClassNames(state),
});

const mapDispatchToProps = {
  setActiveCardClassName: collectionSetActiveClassNameAction,
};

const ConnectedClassBar = connect(mapStateToProps, mapDispatchToProps)(ClassBar);

export default ConnectedClassBar;
