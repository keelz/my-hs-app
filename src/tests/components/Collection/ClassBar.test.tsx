import * as React from 'react';
import * as enzyme from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import ClassBar from '../../../components/Collection/ClassBar';

const defaultProps = Object.freeze({
  activeClassName: 'DRUID',
  cardClassNames: ['DRUID', 'HUNTER', 'MAGE'],
  setActiveCardClassName: jest.fn(),
});

describe('Collection/ClassBar', () => {
  describe('snapshots', () => {
    it('renders without crashing', () => {
      const wrapper = enzyme.render(<ClassBar {...defaultProps} />);
      expect(renderToJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('integration', () => {
    it('handles button click correctly', () => {
      const props = {
        ...defaultProps,
        setActiveCardClassName: jest.fn(),
      };
      const wrapper = enzyme.mount(<ClassBar {...props} />);
      const buttons = wrapper.find('button');
      buttons.at(1).simulate('click');
      expect(props.setActiveCardClassName).toHaveBeenCalledWith('HUNTER');
    });
  });
});
