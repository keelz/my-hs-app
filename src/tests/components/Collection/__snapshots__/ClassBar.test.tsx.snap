// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection/ClassBar snapshots renders without crashing 1`] = `
<div
  class="Nav-bar Collection-classbar"
  id="card-class-name-nav-bar"
>
  <ul
    class="Nav-bar-list"
  >
    <li
      class="Nav-item"
    >
      <button
        class="btn btn-light active"
      >
        DRUID
      </button>
    </li>
    <li
      class="Nav-item"
    >
      <button
        class="btn btn-light"
      >
        HUNTER
      </button>
    </li>
    <li
      class="Nav-item"
    >
      <button
        class="btn btn-light"
      >
        MAGE
      </button>
    </li>
  </ul>
</div>
`;
